server/src/main/java/org/elasticsearch/search/aggregations/metrics/GlobalOrdCardinalityAggregator.java

```diff
@@ -93,7 +93,10 @@ public GlobalOrdCardinalityAggregator(
 
     @Override
     public ScoreMode scoreMode() {
-        if (field != null && valuesSource.needsScores() == false && maxOrd <= MAX_FIELD_CARDINALITY_FOR_DYNAMIC_PRUNING) {
+        if (this.parent == null
```

Comment:
Ok. I understand this now. It took some reading.

The problem is that we were claiming to only need `TOP_DOCS` in this branch because we thought we were going to take the dynamic pruning approach. But we *can't* take that approach because we have a parent. And if we're not taking that approach we have to be `COMPLETE` or `COMPLETE_NO_SCORES`. And being wrong here doesn't matter a lot of the time - but in `nested`, it does.

Reply:
For the sake of the next reader, can you leave a comment saying that this has to line up with the dynamic pruning branch?

Reply:
Perfect explanation. Added.